pgjdbc/src/test/java/org/postgresql/test/jdbc3/Jdbc3CallableStatementTest.java

```diff
@@ -1008,4 +1011,22 @@ public void testSum() throws SQLException {
     assertEquals("2+3 should be 5 when executed via {?= call mysum(?, ?)}", 5, cs.getInt(1));
   }
 
+  @Test
+  public void testNoParametersWithParentheses() throws SQLException {
+    assumeCallableStatementsSupported();
+    CallableStatement cs = con.prepareCall("{?= call mynoparams()}");
+    cs.registerOutParameter(1, Types.INTEGER);
+    cs.execute();
+    assertEquals("{?= call mynoparam()} should return zero, but did not.", 0, cs.getInt(1));
+  }
+
+  @Test
+  public void testNoParametersWithoutParentheses() throws SQLException {
+    assumeCallableStatementsSupported();
+    CallableStatement cs = con.prepareCall("{?= call mynoparams}");
```

Comment:
Just in case: https://github.com/pgjdbc/pgjdbc/blob/63ee60e2ad93441ff55dab08213cc09815b614e0/pgjdbc/src/test/java/org/postgresql/test/jdbc2/AutoRollbackTestSuite.java#L128-L132

The idea is you can use a couple of nested `for` loops to generate all kinds of permutations.